nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb

```diff
@@ -46,16 +48,20 @@ def self.dependency_url_search_cache
           CacheManager.cache("dependency_url_search_cache")
         end
 
-        def initialize(dependency_files:, credentials:)
+        def initialize(dependency_files:, credentials:, repo_contents_path: nil)
           @dependency_files       = dependency_files
           @credentials            = credentials
+          @repo_contents_path     = repo_contents_path
         end
 
-        def dependency_set(project_file:)
+        def dependency_set(project_file:, visited_project_files: Set.new)
```

Comment:
Similar issue here where we would end up in an infinite loop. Each iteration had a clear cache so we would end up looping over the same set of files after expanding the wildcard. Could this cause problems in large projects?